### ISSUE TITLE
[Platform Console UI] Onboarding to Kubernetes (EKS)

### ISSUE THREAD

**Issue by jsgarmon** [department-of-veterans-affairs/va.gov-team#35784]
## Description
The Console UI team would like to deploy to EKS dev/staging. They currently just have a docker container, so will need:
- ECR (terraform)
- service account (terraform)
- manifest (ArgoCD)
- possibly other resources (e.g, persistent database or data store)?

## Background/context 
https://dsva.slack.com/archives/CBU0KDSB1/p1642770950105100

---
## Tasks
- [ ] Determine what AWS resources are needed: add them to the Terraform code
- [ ] Setup deployment pipeline in ArgoCD

## Acceptance Criteria 
- [ ] Their application is running

**Comment by ericvoshall** [comment]
Some additional information:
- Repo: https://github.com/department-of-veterans-affairs/backstage-test
- We will need to use PostgreSQL for a persistent database.

**Comment by ericvoshall** [comment]
@jsgarmon - is there anything I can work on or get in place on my end while your team works on this?

**Comment by jsgarmon** [comment]
Does everyone on your team that will need to access AWS resources for debugging have AWS access already? If not, the approval process can be lengthy so the sooner the better to get started on that.

**Comment by ericvoshall** [comment]
Do you know which AWS resources specifically we should request access to?

**Comment by jsgarmon** [comment]
Since this is going to EKS and you'll need a database, you should just need EKS and RDS.

**Comment by ericvoshall** [comment]
Is there any estimate on completion for this?

**Comment by jsgarmon** [comment]
I'll leave it to @jhouse-solvd to set the priority level on this. Likely related to #34268.

We haven't yet onboarded a team directly to EKS, so this might take some figuring out as we flesh out our process for that.

**Comment by td-usds** [comment]
is this about granting permissions to the console UI team or something else?
If it is about granting them those permissions, then that doesn't seem right to me.
The ConsoleUI team should be able to just deliver a container that gets put into the cluster by Ops. The Console UI team should not be able to touch that cluster.

And if what the Console UI team's deliverable (in this case/iteration/...) is just a react or vue web-app, then their 'deployment' should really be nothing more but "make the CI/CD pipeline put it up in this S3 bucket"

**Comment by jsgarmon** [comment]
This ticket is about doing the work to get their app deployed to EKS. This is work for the infrastructure team to do.

I would argue the app teams should at least be able to reach into dev and staging pods for their application for debugging and testing purposes.

**Comment by td-usds** [comment]
I ack the first part (this is about work for the infrastructure team to do)
I argue the opposite for the second part if what you mean is "as a general state of things, dev team shouldn't have to request get elevated permissions to reach into the dev or staging pods"
I'm not saying they _can't_ get that access, I'm saying they shouldn't have that access by default and should only be given that access when requested for the requested time window...

**Comment by jsgarmon** [comment]
I don't know what you mean by "elevated permissions". I consider access to dev and staging environments to be essential for app teams to do their work. I'm of the belief that app teams should own their products, and they can't own them if they can't access them. Not my call to make, though. I'll let you, @f1337 , and @jhouse-solvd work out what our default level of permissioning should be for apps running in EKS.

**Comment by td-usds** [comment]
I think I can be persuaded on the dev-cluster side... although I still have differences of opinion on that... let me noodle a bit more on this...

**Comment by ericvoshall** [comment]
Hi, any update on this?

**Comment by lhelmer** [comment]
@td-usds Just checking in on your noodling

**Comment by jhouse-solvd** [comment]
Recent planning conversation in Slack [here](https://dsva.slack.com/archives/C02SLSSGSBX/p1643745349347059?thread_ts=1643220638.505119&cid=C02SLSSGSBX)

tl;dr - The infra team is refining the onboarding process and will reach out to clarify requirements in the next couple of days

**Comment by jhouse-solvd** [comment]
@ericvoshall - I've shared the updated ["Application Onboarding Template"](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=&labels=operations%2C+devops%2C+eks%2C+kubernetes%2C+infrastructure%2C+infrastructure-request&template=kubernetes-onboarding-template.md&title=%5BInsert+app+name+here%5D+-+Application+onboarding+infrastructure+request) with Lee & John. 

It would be great to link this issue to the onboarding issue that you will create so that we can easily refer to comments from TD. We're moving this issue to review, and will close once that's done. We'll track progress on getting set up in EKS in the new issue.

**Comment by jchapin** [comment]
@jhouse-solvd, when the time is right, when would @ericvoshall and I get access to push up a branch for our config to... https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests?

Or is there another procedure for getting the configs?

**Comment by jsgarmon** [comment]
FYI, the ECR repo is `008577686731.dkr.ecr.us-gov-west-1.amazonaws.com/dsva/platform-console-ui`

**Comment by jhouse-solvd** [comment]
@jchapin @jsgarmon - Is it okay if we track this work in #36494? If so, I can link this request there for posterity's sake, and then close it if that's alright. 

I believe there's equal and better info there...

**Comment by jsgarmon** [comment]
Yep, that makes sense. This ticket is already referenced in that one, so closing now so that we don't have duplicate tickets open.
